### ISSUE TITLE
SXC Slaughter WML errors about unstore_unit leader

### ISSUE THREAD

**Issue by stevecotton** [TeamSurvivalXtreme/SXCollection#33]
During the AI turn, there are lots of warnings about

    Invalid WML found: [unstore_unit]: variable 'leader' doesn't contain unit data

That seems to be from the code for limiting SXC bosses to a maximum of 30 recruits. While that limit wouldn't be right for Slaughter, the code needs updating. The warning seems to be caused that SXC_Slaughter.cfg doesn't assign the role "big_boss" to any leaders.

**Comment by stevecotton** [comment]
Affects SXC_Rush too, and probably many others

**Comment by stevecotton** [comment]
Although there is a bug causing these warning, all my comments about "big_boss" are wrong - that's set automatically for all AI sides in the normal SXC macros.

**Comment by stevecotton** [comment]
Thanks @Jordy-Jordan for the hint that it involves the fake-side's recruit. I think I now understand the bug, but IIUC it's one that affects the balance of all maps that have fake sides, even those that don't trigger the warning. So any fix to it (other than fixing it in a way that preserves the same behaviour) will need the maps to be rebalanced.

Suppose a fake side whose units are moved by AI 8. This side's recruitment is done during the "new turn" event, which doesn't have a side, but during which it seems `$side_number` can be 1. Then (again IIUC, I haven't debugged through this):

- The new unit is created in `SXC_ENEMY_FAKE_SIDE_TURN`, belonging to side 8.
- The `recruit` event in `SXC_ENEMY_UPGRADE` triggers
    - it will successfully upgrade the unit
    - it will then increment `recruits_$side_number|`, a.k.a. `recruits_1`
    - it will fail to find a `big_boss` on side 1, which leads to the error message

However, killing units on the fake side will decrement `recruits_8`, increasing the recruit limit for the normal boss of side 8.

**Comment by Jordy-Jordan** [comment]
Right, I've even tried to manually add a 10th side to substitute the fake side with
`Invalid WML found: [unstore_unit]: variable 'leader' doesn't contain unit data`

**Comment by Jordy-Jordan** [comment]
I also think the only way to fix it is to rebalance the maps

**Comment by stevecotton** [comment]
Okay, there's less problems than I thought.

The fake recruit code will set `unit.role=fake_recruits`, which means that they don't affect the recruit limits when killed. And this seems to be working, even for recruit events that caused a WML error.

**Comment by Jordy-Jordan** [comment]
Yeah, that should work

**Comment by stevecotton** [comment]
Sorry for forgetting that 10-side support isn't in yet, I've opened #38 for cleaning it up and merging it.

**Comment by Jordy-Jordan** [comment]
Was there a missing "=" in `) role fake_recruits}` on line  1657 from SXC_unit_macros.cfg ?

**Comment by stevecotton** [comment]
No missing "=", it's the three-argument `MODIFY_UNIT FILTER VAR VALUE` macro.

**Comment by Jordy-Jordan** [comment]
thx
